### PR TITLE
feat: swagger migration from spring-fox to spring-doc 

### LINKED ISSUE #118: [P1] Migration swagger docs from spring-fox to spring-doc
[springfox](https://github.com/springfox/springfox) is no longer maintained and the order of parameters are not correctly ordered.

This should be revised in order to make it maintainable

### PR DESCRIPTION
**Pull request by seiwonpark**
**Edited**
There were missing docs for responses.

<br/>   

**Before (spring-fox)**

Screenshot 2023-04-22 at 4 18 03 PM

<br/>   

**After (spring-doc)**

Screenshot 2023-04-22 at 4 18 30 PM

Closes #118

### CODE REVIEW

File: api/src/main/java/com/thesurvey/api/dto/request/AnsweredQuestionRequestDto.java [hunk 1]
```diff
@@ -9,13 +10,16 @@
 import lombok.Builder;
 import lombok.Getter;
 
+// FIXME: this should be fixed since answers should only have one of 4 question types
```
Comment (bezzang-dev) [inline api/src/main/java/com/thesurvey/api/dto/request/AnsweredQuestionRequestDto.java:13]: Could you provide further explanation on the part that needs to be fixed??

Comment (seiwonpark) [reply api/src/main/java/com/thesurvey/api/dto/request/AnsweredQuestionRequestDto.java:13]: Ah it was about explanation. This shows everything (all question types) in a json form. But we need specific one answer for that question type.

Comment (seiwonpark) [reply api/src/main/java/com/thesurvey/api/dto/request/AnsweredQuestionRequestDto.java:13]: No problem on feature but only in swagger document, it shows all 4 answers in one question bank.

Comment (bezzang-dev) [reply api/src/main/java/com/thesurvey/api/dto/request/AnsweredQuestionRequestDto.java:13]: I got it. Thank you for the explanation!

**Review by bezzang-dev — APPROVED** [review]
LGTM! Thanks!

**Review by bezzang-dev — APPROVED** [review]
Thank you for the refactor including api documentation. LGTM!

**Pull request merged**